x/programs/rust/examples/lottery/src/lib.rs

```diff
@@ -8,19 +7,18 @@ static TOKEN_PROGRAM_NAME: &str = "token_contract";
 
 /// Initializes the program.
 #[expose]
-fn init_program() -> i64 {
+fn init(_: Context) -> bool {
     // Initialize the program with no fields
-    Program::new().into()
+    true
 }
```

Comment:
I think the fact that we end up with a hard coded bool here speaks to the fact that init should not be a requirement.